ebmlite/core.py

```diff
@@ -811,6 +811,24 @@ def __repr__(self):
         return "<%s %r at 0x%08X>" % (self.__class__.__name__, self.name,
                                       id(self))
 
+    def __enter__(self):
+        """
+        Enter context manager for this document.
+
+        Expects to clean up `self.stream` object; raises an error if stream
+        object needs no such cleanup.
+        """
+        if not hasattr(self.stream, 'close'):
```

Comment:
Are there any random-access stream-like objects (in Python itself or in popular packages) without `close()`?

Reply:
I assumed so from this comment, but maybe not. https://github.com/MideTechnology/ebmlite/blob/development/ebmlite/core.py#L815

Reply:
addressed in 9f7d5c696e5d39c449ec9442a12e259f310fd3b9